site/lazy-queues.md

```diff
@@ -0,0 +1,244 @@
+
+A `default` queue will keep messages in memory until memory pressure forces them to be paged out to disk.
+This will result in delayed disk I/O which can be significant since more data will need to be written to disk at once.
+
+### RAM Utilization
+
+While it's hard to give numbers that make sense for every use case,
+this is a simplistic test that showcases the difference in RAM utilization between a `default` &amp; a `lazy` queue:
+
+| Number of messages | Message body size | Message type | Producers | Consumers |
+| -                  | -                 | -            | -         | -         |
+| 1,000,000          | 1,000 bytes       | persistent   | 1         | 0         |
+
+The RAM utilization for `default` &amp; `lazy` queues **after** ingesting the above messages:
+
+| Queue type | Queue process memory | Messages in memory | Memory used by messages | Node memory |
+| -          | -                    | -                  | -                       | -           |
+| `default`  | 257 MB               | 386,307            | 368 MB                  | 734 MB      |
+| `lazy`     | 159 KB               | 0                  | 0                       | 0           |
+
+Both queues persisted 1,000,000 messages and used 1.2 GB of disk space.
+
+`default` queue test:
+
+<pre class="sourcecode bash">
+# Start a temporary RabbitMQ node:
+#
+#       export RABBITMQ_NODENAME=default-queue-test
+#       export RABBITMQ_MNESIA_BASE=/tmp
+#       export RABBITMQ_LOG_BASE=/tmp
+#       rabbitmq-server &amp;
+#
+# (the last command will fail if there is another RabbitMQ node already running)
+
+# In a https://github.com/rabbitmq/rabbitmq-perf-test clone, run:
+make run ARGS="-y 0 -s 1000 -f persistent -C 1000000 -u default -ad false"
+
+# Queue stats:
+rabbitmqctl list_queues name arguments memory messages_ram message_bytes_ram messages_persistent message_bytes_persistent
+Timeout: 60.0 seconds ...
+Listing queues for vhost / ...
+default	[]	417421592	386307	386307000	1000000	1000000000
+
+# Node memory stats
+rabbitmqctl status | grep rss,
+      {total,[{erlang,1043205272},{rss,770306048},{allocated,1103822848}]}]},
+
+# Stop our temporary RabbitMQ node &amp; clean all persistent files
+#
+#       rabbitmqctl shutdown
+#       rm -fr /tmp/{log,$RABBITMQ_NODENAME*}
+</pre>
+
+The `lazy` queue test is almost the same, these are the differences:
+
+<pre class="sourcecode bash">
+# Use a different RABBITMQ_NODENAME
+#
+#       export RABBITMQ_NODENAME=lazy-queue-test
+
+# In a https://github.com/rabbitmq/rabbitmq-perf-test clone, run:
+make run ARGS="-y 0 -s 1000 -f persistent -C 1000000 -u lazy -qa x-queue-mode=lazy -ad false"
+</pre>
+
+**Note that this was a very simplistic test.**
+**Please make sure to run your own benchmarks.**
+
+### Converting between queue modes
+
+If we need to convert a `default` queue into a `lazy` one,
+then we will suffer the same performance impact as when a queue needs to page messages to disk.
```

Comment:
When converting a `default` queue into a `lazy` one, the operation will suffer the same performance impact as when a queue needs to page messages to disk.